src/components/modal/modal.test.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+import Modal from "./modal.component";
+import useScrollBlock from "../../hooks/__internal__/useScrollBlock";
+
+jest.mock("../../hooks/__internal__/useScrollBlock");
+const allowScroll = jest.fn();
+const blockScroll = jest.fn();
+
+const mockedUseScrollBlock = useScrollBlock as jest.MockedFunction<
+  typeof useScrollBlock
+>;
+
+mockedUseScrollBlock.mockReturnValue({
+  allowScroll,
+  blockScroll,
+});
+
+let onCancelFn: jest.Mock,
+  onCancelFnTwo: jest.Mock,
+  addEventListenerSpy: jest.SpyInstance,
+  removeEventListenerSpy: jest.SpyInstance;
+
+describe("event listeners", () => {
+  beforeEach(() => {
+    addEventListenerSpy = jest.spyOn(document, "addEventListener");
+    removeEventListenerSpy = jest.spyOn(document, "removeEventListener");
+  });
+
+  afterEach(() => {
+    addEventListenerSpy.mockRestore();
+    removeEventListenerSpy.mockRestore();
+  });
+
+  it("binds the key event listener to the document on mount", () => {
```

Comment:
**suggestion** (non-blocking): making this comment non-blocking as I'm not sure how easy this will be, but I think it should be possible - all these tests for adding/removing event listeners seem too focused on implementation. The important thing is that, once the modal is mounted, it responds to key presses, and then doesn't once unmounted. I would have thought that we can test for that by simulating actual user interactions, rather than the implementation details of when we call `addEventListener` and `removeEventListener`.